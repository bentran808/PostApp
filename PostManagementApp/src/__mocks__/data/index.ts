const initialUser = {
  id: '',
  avatar: '',
  name: '',
  gender: '',
  email: '',
  phone: '',
  address: '',
  role: ''
};

const mockAdmin = {
  id: '621c4188dece66a40dad2951',
  avatar: 'https://picsum.photos/200',
  name: 'Admin',
  gender: 'male',
  email: 'admin@admin.com',
  phone: '+84 (919) 507231',
  address: '199 Bergen Street, Bowie, Oregon, 7933',
  role: 'admin'
};

const mockUser = {
  id: '621c41883112f1c216433dcf',
  avatar: 'https://picsum.photos/200',
  name: 'Mcintyre Morrison',
  gender: 'male',
  email: 'test1@email.com',
  phone: '+84 (838) 539391',
  address: '313 Menahan Street, Aurora, North Dakota, 9139',
  role: 'user'
};

const mockPost = {
  author: mockAdmin,
  company: 'bb',
  year: 2222,
  type: 'b',
  status: true,
  price: 23333111,
  address: '199 Bergen Street, Bowie, Oregon, 7933',
  title: 'bb update',
  description: 'bb',
  photos: [
    {
      height: 480,
      uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      width: 480,
      fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      type: 'image/jpeg',
      fileSize: 6689
    },
    {
      height: 480,
      uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      width: 480,
      fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
      type: 'image/jpeg',
      fileSize: 6689
    }
  ],
  pending: false,
  createdAt: 1647507076963,
  updatedAt: 1647511146749,
  id: 2
};

const mockLikes = [
  {
    author: mockAdmin,
    postId: 2,
    createdAt: 1647509052801,
    updatedAt: 1647509052801,
    id: 1
  }
];

const mockCommentsList = [
  {
    author: mockAdmin,
    postId: 2,
    content: 'test',
    createdAt: 1647509052801,
    updatedAt: 1647509052801,
    id: 1
  },
  {
    author: {
      id: '621c4188e8365653f9326450',
      avatar: 'https://picsum.photos/200',
      name: 'Frost Spence',
      gender: 'male',
      email: 'frostspence@entogrok.com',
      phone: '+84 (859) 508289',
      address: '745 Blake Avenue, Belmont, District Of Columbia, 4099',
      role: 'user'
    },
    postId: 2,
    content: 'test2',
    createdAt: 1647509052801,
    updatedAt: 0,
    id: 2
  }
];

const mockNewPost = {
  author: mockAdmin,
  company: 'qwe',
  year: 2021,
  type: 'qwe',
  status: true,
  price: 10000,
  address: 'qwe',
  title: 'qwe',
  description: 'qwe',
  photos: [],
  pending: true
};

const mockEditedPost = {
  '2': mockPost,
  '3': {
    id: 3,
    ...mockNewPost
  }
};

const mockApprovedPost = {
  '2': mockPost,
  '3': {
    id: 3,
    ...mockNewPost,
    pending: false
  }
};

const mockLike = {
  author: mockAdmin,
  postId: 2,
  createdAt: 1647509052801,
  updatedAt: 1647509052801,
  id: 2
};

const mockNewComment = {
  author: mockAdmin,
  postId: 2,
  content: 'abc',
  createdAt: 1647509052801,
  updatedAt: 1647509052801,
  id: 3
};

const mockPhotos = [
  {
    fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    fileSize: 6689,
    height: 480,
    type: 'image/jpeg',
    uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    width: 480
  },
  {
    fileName: 'rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    fileSize: 6689,
    height: 480,
    type: 'image/jpeg',
    uri: 'file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg',
    width: 480
  }
];

export {
  initialUser,
  mockAdmin,
  mockUser,
  mockPost,
  mockLikes,
  mockCommentsList,
  mockNewPost,
  mockEditedPost,
  mockApprovedPost,
  mockLike,
  mockNewComment,
  mockPhotos
};
