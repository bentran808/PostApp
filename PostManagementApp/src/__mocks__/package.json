{
    "name": "__mock__",
    "private": true
}