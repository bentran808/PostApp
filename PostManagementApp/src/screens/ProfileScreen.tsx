import React, {useContext} from 'react';
import {StyleSheet, View} from 'react-native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import FormButton from '../components/FormButton';
import {CounterContext} from '../navigation/CounterContext';

type LoginNavigationProp = NativeStackNavigationProp<
    RootStackParamList,
    'Login'
>;

interface LoginProps {
    navigation: LoginNavigationProp;
}

const ProfileScreen = ({navigation}: LoginProps) => {
    const {logoutUser} = useContext(CounterContext);
    return (
        <View style={styles.body}>
            <FormButton
                title="Logout"
                onPress={() => {
                    logoutUser();
                    navigation.navigate('Login');
                }}
            />
        </View>
    );
};

const styles = StyleSheet.create({
    body: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        paddingHorizontal: 20
    }
});

export default ProfileScreen;
