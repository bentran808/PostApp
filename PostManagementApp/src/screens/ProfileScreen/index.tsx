import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Button, Title } from 'react-native-paper';
import { colors } from '../../constants/colors';
import { useAppDispatch, useAppSelector } from '../../hooks';
import { authActions, selectCurrentUser } from '../../redux/slices';

type LoginNavigationProp = NativeStackNavigationProp<RootStackParamList>;

interface LoginProps {
    navigation: LoginNavigationProp;
}

const ProfileScreen = ({ navigation }: LoginProps) => {
    const dispatch = useAppDispatch();
    const currentUser = useAppSelector((state) => selectCurrentUser(state));

    return (
        <View style={styles.body}>
            <Title>{currentUser.name}</Title>
            <Button
                mode="contained"
                color={colors.royalBlue}
                uppercase={false}
                onPress={() => {
                    dispatch(authActions.logout());
                    navigation.navigate('Login');
                }}
            >
                Logout
            </Button>
        </View>
    );
};

const styles = StyleSheet.create({
    body: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        paddingHorizontal: 20
    }
});

export default ProfileScreen;
