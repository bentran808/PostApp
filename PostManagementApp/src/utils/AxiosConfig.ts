import axios from 'axios';

export const axiosInstance = axios.create({
    baseURL: 'http://192.168.56.1:3000/',
    headers: {
        'Content-Type': 'application/json'
    }
});
