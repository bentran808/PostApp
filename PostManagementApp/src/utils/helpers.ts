import AsyncStorage from '@react-native-async-storage/async-storage';

const sortDesc = (arr: Post[] = []) => {
  return arr.sort((a, b) => ((a.id || 0) < (b.id || 0) ? 1 : -1));
};

const formatPrice = (price: Number) => {
  return price.toLocaleString('it-IT', {
    style: 'currency',
    currency: 'VND'
  });
};

const getApprovedPosts = (arr: Post[] = []) => arr.filter((item) => !item.pending);

const setAccessToken = async (access_token: string) =>
  await AsyncStorage.setItem('access_token', access_token);

const getAccessToken = async () => await AsyncStorage.getItem('access_token');

const removeAccessToken = async () => await AsyncStorage.removeItem('access_token');

export {
  sortDesc,
  formatPrice,
  getApprovedPosts,
  setAccessToken,
  getAccessToken,
  removeAccessToken
};
