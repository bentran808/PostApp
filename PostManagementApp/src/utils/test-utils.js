import React from 'react';
import { Provider } from 'react-redux';
import { Provider as PaperProvider } from 'react-native-paper';
import { render } from '@testing-library/react-native';
import createSagaMiddleware from 'redux-saga';
import { configureStore } from '@reduxjs/toolkit';
import authReducer from 'redux/slices/authSlice';
import postReducer from 'redux/slices/postSlices';

const sagaMiddleware = createSagaMiddleware();
const AllTheProviders = ({ children }) => {
  const store = configureStore({
    reducer: {
      auth: authReducer,
      post: postReducer
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        serializableCheck: false
      }).concat(sagaMiddleware)
  });
  return (
    <Provider store={store}>
      <PaperProvider>{children}</PaperProvider>
    </Provider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
