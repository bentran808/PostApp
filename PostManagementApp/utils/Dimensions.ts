import {Dimensions} from 'react-native';

const windowHeight = Dimensions.get('window').height;

export {windowHeight};
