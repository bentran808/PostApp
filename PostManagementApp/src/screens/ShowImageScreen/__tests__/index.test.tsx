import React from 'react';
import ProfileScreen from '..';
import { render } from 'utils/test-utils';
import { mockPhotos } from '__mocks__/data';

describe('Test Show Image Screen', () => {
  const route = {
    params: {
      photos: mockPhotos
    }
  };

  test('should render correctly', () => {
    const { toJSON } = render(<ProfileScreen route={route} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
