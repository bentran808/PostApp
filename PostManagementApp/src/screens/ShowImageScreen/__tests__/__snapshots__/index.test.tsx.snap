// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Show Image Screen should render correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#000000",
        "height": "100%",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "height": 300,
          "overflow": "hidden",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        bounces={false}
        data={
          Array [
            "file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg",
            "file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg",
          ]
        }
        decelerationRate="fast"
        getItem={[Function]}
        getItemCount={[Function]}
        getItemLayout={[Function]}
        horizontal={true}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        onViewableItemsChanged={[Function]}
        pagingEnabled={true}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "height": "100%",
            "width": 750,
          }
        }
        viewabilityConfig={
          Object {
            "viewAreaCoveragePercentThreshold": 60,
            "waitForInteraction": false,
          }
        }
        viewabilityConfigCallbackPairs={
          Array [
            Object {
              "onViewableItemsChanged": [Function],
              "viewabilityConfig": Object {
                "viewAreaCoveragePercentThreshold": 60,
                "waitForInteraction": false,
              },
            },
          ]
        }
      >
        <View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "height": "auto",
                  "width": 750,
                }
              }
            >
              <Image
                onLoad={[Function]}
                resizeMode="cover"
                source={
                  Object {
                    "uri": "file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg",
                  }
                }
                style={
                  Object {
                    "height": "100%",
                    "overflow": "hidden",
                    "resizeMode": "cover",
                    "width": 750,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#D3D3D3",
                    "height": "100%",
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": 750,
                  }
                }
              >
                <ActivityIndicator
                  color="#000"
                  size="large"
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "height": "auto",
                  "width": 750,
                }
              }
            >
              <Image
                onLoad={[Function]}
                resizeMode="cover"
                source={
                  Object {
                    "uri": "file:///data/user/0/com.PostManagementApp/cache/rn_image_picker_lib_temp_283c7c3b-b014-4ef4-b95a-c092bcefe2aa.jpg",
                  }
                }
                style={
                  Object {
                    "height": "100%",
                    "overflow": "hidden",
                    "resizeMode": "cover",
                    "width": 750,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#D3D3D3",
                    "height": "100%",
                    "justifyContent": "center",
                    "position": "absolute",
                    "width": 750,
                  }
                }
              >
                <ActivityIndicator
                  color="#000"
                  size="large"
                />
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 25,
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "position": "absolute",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#FFF",
                "borderColor": "#FFF",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 20,
                "marginRight": 7,
                "width": 20,
              },
              Object {
                "backgroundColor": "#000",
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#FFF",
                "borderColor": "#FFF",
                "borderRadius": 50,
                "borderWidth": 2,
                "height": 20,
                "marginRight": 7,
                "width": 20,
              },
              false,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
